src/Microsoft.AspNetCore.Server.Kestrel/Http/MessageBody.cs

```diff
@@ -120,7 +120,7 @@ public Task Consume(CancellationToken cancellationToken = default(CancellationTo
             var contentLength = headers.HeaderContentLength.ToString();
             if (contentLength.Length > 0)
             {
-                return new ForContentLength(keepAlive, int.Parse(contentLength), context);
+                return new ForContentLength(keepAlive, long.Parse(contentLength), context);
```

Comment:
@cesarbs is now going to have to rebase on top of your chunked requests changes, so it should be there as long as he doesn't remove it :wink: